### PR TITLE
PI-1708 log language in SDKUI

### PR DESCRIPTION
**Pull request by chintan-soni-cko**
## Proposed changes
1 Added User's device default language to paymentFormPresented's event data
2 Added unit tests to support changes

## Types of changes

What types of changes does your code introduce to frames-android?
_Put an `x` in the boxes that apply_

* [ ] Bugfix (non-breaking change which fixes an issue)
* [X] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

* [X] Lint and unit tests pass locally with my changes
* [X] I have added tests that prove my fix is effective or that my feature works
* [ ] I have added necessary documentation (if appropriate)

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: android-sdk/build.gradle [hunk 1]
```diff
@@ -50,6 +54,10 @@ dependencies {
     testImplementation 'org.mockito:mockito-core:3.5.13'
     androidTestImplementation 'androidx.test.ext:junit:1.1.2'
     androidTestImplementation 'androidx.test.espresso:espresso-core:3.3.0'
+    testImplementation 'org.mockito:mockito-inline:4.4.0'
+    // Added for mocking android components in unit testing
+    testImplementation 'org.robolectric:robolectric:4.7.3'
```
Comment (harry-brown-cko) [inline android-sdk/build.gradle:59]: This is a new dependency and test runner. We should discuss as a team and come to an agreement on new dependencies and test practices

File: android-sdk/src/test/java/com/checkout/android_sdk/PaymentFormTest.kt [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+
+    @Before
+    internal fun setUp() {
+        paymentForm = PaymentForm(ApplicationProvider.getApplicationContext())
+        sdkLoggerMock = mock(CheckoutEventLogger::class.java)
+        mockFramesLogger = mock(FramesLogger::class.java)
+        mockFramesLogger.initialise(ApplicationProvider.getApplicationContext(),
+            Environment.SANDBOX,
+            sdkLoggerMock)
```
Comment (harry-brown-cko) [inline android-sdk/src/test/java/com/checkout/android_sdk/PaymentFormTest.kt:30]: do we need to call initialise here? FramesLogger has been mocked

File: android-sdk/src/main/java/com/checkout/android_sdk/CheckoutAPIClient.java [hunk 3]
```diff
@@ -83,7 +83,8 @@ public CheckoutAPIClient(@NonNull Context context, @NonNull String key, @NonNull
         this.mKey = key;
         this.mEnvironment = environment;
         this.mLogger = framesLogger;
-        //Send checkoutApiClientInitialisedEvent on initialization of CheckoutAPIClient
+       
+      //Send checkoutApiClientInitialisedEvent on initialization of CheckoutAPIClient
```
Comment (harry-brown-cko) [inline android-sdk/src/main/java/com/checkout/android_sdk/CheckoutAPIClient.java:87]: why is this change necessary?

File: android-sdk/src/main/java/com/checkout/android_sdk/logging/FramesLoggingEventDataProvider.kt [hunk 4]
```diff
@@ -1,10 +1,21 @@
 package com.checkout.android_sdk.logging
-
-import com.checkout.android_sdk.Utils.Environment
 import com.checkout.eventlogger.domain.model.MonitoringLevel
+import com.checkout.android_sdk.Utils.Environment
+import java.util.*
```
Comment (harry-brown-cko) [inline android-sdk/src/main/java/com/checkout/android_sdk/logging/FramesLoggingEventDataProvider.kt:4]: is this even used?

File: android-sdk/src/test/java/com/checkout/android_sdk/PaymentFormTest.kt (hunk 2 above)
Comment (chintan-soni-cko) [reply android-sdk/src/test/java/com/checkout/android_sdk/PaymentFormTest.kt:30]: No, we don't require it. I will remove it.

Comment (chintan-soni-cko) [reply android-sdk/src/test/java/com/checkout/android_sdk/PaymentFormTest.kt:30]: No, we don't require it. I will remove it.

File: android-sdk/build.gradle (hunk 1 above)
Comment (chintan-soni-cko) [reply android-sdk/build.gradle:59]: Reason for adding 2 dependencies:
1. Mockito inline library: used to mock the final or library class. For instance, CheckoutEventLogger class. with core library, it is going to throw as a could not spy final, static methods or class.
2. Robolectric Runner:  It is used for test doubles (standard unit testing framework). We are required to unit test code with the mocking of android components. For instance, to inject Paymentform class in unit test passed android context, to unit test Webview. It simulated android components in JVM without running on a real-time device.
Reference: http://robolectric.org/#:~:text=Robolectric%20handles%20inflation%20of%20views,do%20on%20a%20real%20device.

File: android-sdk/src/main/java/com/checkout/android_sdk/logging/FramesLoggingEventDataProvider.kt (hunk 4 above)
Comment (chintan-soni-cko) [reply android-sdk/src/main/java/com/checkout/android_sdk/logging/FramesLoggingEventDataProvider.kt:4]: Import of java.util is for a Locale instance to pass the language details in events.

File: android-sdk/src/main/java/com/checkout/android_sdk/logging/FramesLoggingEventDataProvider.kt [hunk 5]
```diff
@@ -1,10 +1,21 @@
 package com.checkout.android_sdk.logging
-
-import com.checkout.android_sdk.Utils.Environment
 import com.checkout.eventlogger.domain.model.MonitoringLevel
+import com.checkout.android_sdk.Utils.Environment
```
[the remark this answers is not in the document]
Comment (chintan-soni-cko) [reply android-sdk/src/main/java/com/checkout/android_sdk/logging/FramesLoggingEventDataProvider.kt:3]: This has been reflected as a line change while merging. I will take care of it.

File: android-sdk/src/main/java/com/checkout/android_sdk/logging/FramesLoggingEventDataProvider.kt (hunk 4 above)
Comment (harry-brown-cko) [reply android-sdk/src/main/java/com/checkout/android_sdk/logging/FramesLoggingEventDataProvider.kt:4]: can we just have `import java.util.Locale` instead?

Comment (chintan-soni-cko) [reply android-sdk/src/main/java/com/checkout/android_sdk/logging/FramesLoggingEventDataProvider.kt:4]: It's by default common import for all java utils classes. We can add custom imports for a specific class. However, when we add any more values to events which have dependencies on this utils class, we don't require to import it manually every time.

Comment (harry-brown-cko) [reply android-sdk/src/main/java/com/checkout/android_sdk/logging/FramesLoggingEventDataProvider.kt]: Yes you're right, however this reduces visibility on which classes are being imported and used. There is also the possibility of collision in class names if imported with a wildcard.

As far as best practices go for Java/Kotlin, my understanding is that you should default to importing specific classes until you reach a certain amount from that package, eg. 4, so that the wildcard will save you lines of code.

See [this stackoverflow article](https://stackoverflow.com/questions/187453/import-package-vs-import-package-specifictype) for a discussion on it.

**Review by harry-brown-cko — APPROVED** [review]
